okhttp/src/commonJvmAndroid/kotlin/okhttp3/CompressionInterceptor.kt

```diff
@@ -0,0 +1,150 @@
+  ) : this(
+    algorithms.associateWithTo(LinkedHashMap()) {
+      null
+    },
+  )
+
+  internal val acceptEncoding =
+    if (algorithms.isEmpty()) {
+      "identity"
+    } else {
+      algorithms
+        .map { (algorithm, priority) ->
+          if (priority != null) {
+            "${algorithm.encoding};q=$priority"
+          } else {
+            algorithm.encoding
+          }
+        }.joinToString(separator = ", ")
+    }
+
+  private val algorithmIndex = algorithms.keys.toTypedArray()
+
+  override fun intercept(chain: Interceptor.Chain): Response =
+    if (chain.request().header("Accept-Encoding") == null) {
+      val request =
+        chain
+          .request()
+          .newBuilder()
+          .header("Accept-Encoding", acceptEncoding)
+          .build()
+
+      val response = chain.proceed(request)
+
+      decompress(response)
+    } else {
+      chain.proceed(chain.request())
+    }
+
+  /**
+   * Returns a decompressed copy of the Response, typically via a streaming Source.
+   * If no known decompression or the response is not compressed, returns the response unmodified.
+   */
+  fun decompress(response: Response): Response {
+    if (!response.promisesBody()) {
+      return response
+    }
+    val body = response.body
+    val encoding = response.header("Content-Encoding") ?: return response
+
+    val algorithm = lookupDecompressor(encoding) ?: return response
+
+    val decompressedSource = algorithm.decompress(body.source()).buffer()
+
+    return response
+      .newBuilder()
+      .removeHeader("Content-Encoding")
+      .removeHeader("Content-Length")
+      .body(decompressedSource.asResponseBody(body.contentType(), -1))
+      .build()
+  }
+
+  internal fun lookupDecompressor(encoding: String): DecompressionAlgorithm? {
+    val algorithm = algorithmIndex.find { it.encoding.equals(encoding, ignoreCase = true) } ?: return null
+
+    if (algorithm == Wildcard || algorithm == Identity) {
+      return null
+    }
+
+    return algorithm
+  }
+
+  /**
+   * A decompression algorithm such as Gzip. Must provide the Accept-Encoding value and decompress a Source.
+   */
+  interface DecompressionAlgorithm {
+    val encoding: String
+
+    fun decompress(compressedSource: BufferedSource): Source
+  }
+
+  companion object {
+    /**
+     * Request "gzip" compression.
+     */
+    val Gzip =
+      object : DecompressionAlgorithm {
+        override val encoding: String = "gzip"
+
+        override fun decompress(compressedSource: BufferedSource): Source = GzipSource(compressedSource)
+      }
+
+    /**
+     * Request the response body to be uncompressed.
+     */
+    val Identity =
+      object : DecompressionAlgorithm {
+        override val encoding: String = "identity"
+
+        override fun decompress(compressedSource: BufferedSource): Source = compressedSource
+      }
+
+    /**
+     * Declare support for any compression encoding scheme. However the response will not be decompressed.
```

Comment:
Why?

Reply:
Will revert to previous commit and land.